src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.Linux.cs

```diff
@@ -21,6 +34,235 @@ private SslStreamCertificateContext(X509Certificate2 target, X509Certificate2[]
             SslContexts = new ConcurrentDictionary<SslProtocols, SafeSslContextHandle>();
         }
 
-        internal static SslStreamCertificateContext Create(X509Certificate2 target) => Create(target, null);
+        internal static SslStreamCertificateContext Create(X509Certificate2 target) =>
+            Create(target, null, offline: false, trust: null, noOcspFetch: true);
+
+        internal bool OcspStaplingAvailable => _ocspUrls is not null;
+
+        partial void SetNoOcspFetch(bool noOcspFetch)
+        {
+            _staplingForbidden = noOcspFetch;
+        }
+
+        partial void AddRootCertificate(X509Certificate2? rootCertificate)
+        {
+            if (IntermediateCertificates.Length == 0)
+            {
+                _ca = rootCertificate;
+            }
+            else
+            {
+                _ca = IntermediateCertificates[0];
+            }
+
+            if (!_staplingForbidden)
+            {
+                // Create the task, let the download finish in the background.
```

Comment:
Can you elaborate on why it's ok for this to be fire-and-forget?

Reply:
Consistent with other platforms, the server side of OCSP stapling is bestish effortish.  On Windows they equivalently fire off an async download, and whenever there's a new session and the download has completed they attach it... if it didn't complete (or it failed), then it isn't.